### PR TITLE
Create swift.yml

### PR DESCRIPTION
**Pull request by compnerd**
Add an initial GitHub workflow to enable a `swift-format` linting action.

### CODE REVIEW

**Comment by compnerd** [comment]
@googlebot I fixed it.

File: Sources/TensorFlow/Core/MixedPrecision.swift [hunk 1]
```diff
@@ -157,7 +157,7 @@ extension Tensor {
   }
 
   /// Promotes a scalar to a tensor with the same device and precision as the given tensor.
-  @differentiable(where Scalar: TensorFlowFloatingPoint)
+  @differentiable( where Scalar: TensorFlowFloatingPoint)
```
Comment (saeta) [inline Sources/TensorFlow/Core/MixedPrecision.swift:160]: Note: seems like it's mis-formatting this at the moment. (I think you're already aware, but wanted to make sure to flag them here.)

File: Sources/TensorFlow/Core/MixedPrecision.swift [hunk 2]
```diff
@@ -206,12 +206,12 @@ extension Tensor {
 #endif
 
 extension Tensor where Scalar: TensorFlowFloatingPoint {
-  @derivative(of: toReducedPrecision)
+  @derivative(of:toReducedPrecision)
```
Comment (saeta) [inline Sources/TensorFlow/Core/MixedPrecision.swift:209]: And here.

File: Sources/TensorFlow/Core/MixedPrecision.swift (hunk 1 above)
Comment (compnerd) [reply Sources/TensorFlow/Core/MixedPrecision.swift:160]: Yeah, @allevato is going to be looking into backporting the fixes for that to the 5.2 release so we can get these correct.

**Comment by compnerd** [comment]
Superseded by #931

**Pull request closed without merging**